src/main/java/com/revature/services/BatchLogicImpl.java

```diff
@@ -10,57 +17,133 @@
+	
 	@Override
 	public Batch getBatchByName(String batchName) {
 		Batch batch = dao.findByNameIgnoreCase(batchName);
+		
+		if (batch == null) {
+			throw new NotFoundException("Batch with name " + batchName + " not found");
+		}
+		
 		return batch;
 	}
 
 	@Override
 	public Batch getBatchById(int batchId) {
 		Batch batch = dao.findOne(batchId);
-		return batch;
+		
+		if (batch == null) {
+			throw new NotFoundException("Batch with id " + batchId + " not found");
 		}
+		
+		return batch;
+	}
 
 	@Override
-	public void deleteBatch(Batch batchName) {
-		dao.delete(batchName);	
+	public String deleteBatch(Integer id) {
+		Batch batch = getBatchById(id);
+		dao.delete(batch);
+		return "Batch: " + id + " - DELETED";
 	}
 	
+	@Override
+	public Batch removePersonsFromBatch(Integer batchId, Integer[] personIds) {
+		
+		validation.validateIntegerArray(personIds);
+		
+		Batch batch = getBatchById(batchId);
+		
+		Set<Person> persons = batch.getPersons();
+		
+		for (Integer personId : personIds) {
+			validation.validateTraineeInBatch(personId, batchId);
+			persons.remove(personLogic.getPersonById(personId));
+		}
+
+		dao.saveAndFlush(batch);
+		entityManager.refresh(batch);
+		
+		return batch;
+	}
 
 	@Override
 	public Page<Batch> getAllBatches(Pageable pageable) {
 		return dao.findAll(pageable);
 	}
 
 	@Override
-	public void createBatch(Batch batchName) {
-		dao.save(batchName);
+	public Batch createBatch(Batch batch) {
+		
+		validation.validateBatchFields(batch);
+		
+		dao.saveAndFlush(batch);
+		entityManager.refresh(batch);
+		return batch;
+	}
+	
+	@Override
+	public Batch addPersonsToBatch(Integer batchId, Integer[] personIds) {
+		
+		validation.validateIntegerArray(personIds);
+		
+		Batch batch = getBatchById(batchId);
+		
+		Set<Person> persons = batch.getPersons();
 		
+		for (Integer personId : personIds) {
+			validation.validateTraineeExists(personId);
+			persons.add(personLogic.getPersonById(personId));
+		}
+
+		dao.saveAndFlush(batch);
+		entityManager.refresh(batch);
+		
+		return batch;
 	}
 
 	@Override
-	public void updateBatch(Batch batchName) {
-		dao.save(batchName);
+	public Batch updateBatch(Batch batchName, Integer id) {
+		
+		Batch batch = getBatchById(id);
+		
+		if(batchName.getName() != null && !batchName.getName().equals("")){
+			batch.setName(batchName.getName());
+		}
 		
+		if(batchName.getPersons() != null){
+			batch.setPersons(batchName.getPersons());
+		}
+		
+		
+		dao.saveAndFlush(batch);
+		entityManager.refresh(batch);
+		return batch;
 	}
 
 	@Override
 	public Page<Person> getAllPeopleByBatchId(Pageable pageable, Integer id) {
-		 Batch batch = dao.findOne(id);
-		 List<Person> personList = batch.getPersons();
-		Page<Person> personPage = new PageImpl<Person>(personList, pageable, personList.size());
+		
+		Batch batch = getBatchById(id);
+		
+		Set<Person> personList = batch.getPersons();
+		Page<Person> personPage = new PageImpl<Person>((new ArrayList(personList)),
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AUselessParenthesesCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS2293)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "personPage". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1488)